### PR TITLE
Create arcs.util.Log

### PR DESCRIPTION
**Pull request by jasonwyatt**
Going to need this logging util to help work through some issues I'm finding with my easy-mode `ArcsSet` work.

### CODE REVIEW

File: src_kt/java/arcs/util/Log.kt [hunk 1]
```diff
@@ -0,0 +1,90 @@
...
+            writer(level, formatter(logIndex.incrementAndGet(), level, throwable, messageBuilder()))
+        }
+    }
+}
+
+private val DEFAULT_LEVEL = Log.Level.Error
+
+private val DEFAULT_FORMATTER: (index: Int, level: Log.Level, throwable: Throwable?, rawMessage: String) -> String =
+    { index, level, throwable, rawMessage ->
+        "$index - ${level.name}: $rawMessage" +
```
Comment (csilvestrini) [inline src_kt/java/arcs/util/Log.kt]: Is there any particular reason you want to log the index here?

Comment (jasonwyatt) [reply src_kt/java/arcs/util/Log.kt]: I think it'd be useful when someone sends you a log, if they reference some line that has a particular message, you could jump directly to that line with grep/cmd+f (even if there are other instances with the same body.

**Comment by Cypher1** [comment]
This seems to have broken build on master: https://travis-ci.org/PolymerLabs/arcs/builds/609067594

Can you look into it @jasonwyatt ?

**Comment by Cypher1** [comment]
This might be the relevant failure, but I don't have a lot of experience with it.

```
log-factory: binding logFactory to level [0]
bazel-out/k8-fastbuild/bin/particles/Native/Wasm/example.h
INFO: From Compiling Kotlin to JS //src_kt/java/arcs/storage:storage-js { kt: 10 }:
warning: module "util-js" is defined in more than one file
ERROR: /home/travis/build/PolymerLabs/arcs/src_kt/java/arcs/storage/api/BUILD:5:1: Compiling Kotlin to JS //src_kt/java/arcs/storage/api:api-js { kt: 1 } failed (Exit 1)
warning: module "util-js" is defined in more than one file
src_kt/java/arcs/storage/api/ArcsSet.kt:8:18: error: unresolved reference: internal
import arcs.crdt.internal.Actor
                 ^
src_kt/java/arcs/storage/api/ArcsSet.kt:9:18: error: unresolved reference: internal
import arcs.crdt.internal.VersionMap
                 ^
src_kt/java/arcs/storage/api/ArcsSet.kt:24:21: error: unresolved reference: referencemode
import arcs.storage.referencemode.RefModeStoreData
                    ^
src_kt/java/arcs/storage/api/ArcsSet.kt:25:21: error: unresolved reference: referencemode
import arcs.storage.referencemode.RefModeStoreOp
                    ^
src_kt/java/arcs/storage/api/ArcsSet.kt:26:21: error: unresolved reference: referencemode
import arcs.storage.referencemode.ReferenceModeStorageKey
                    ^
src_kt/java/arcs/storage/api/ArcsSet.kt:50:33: error: unresolved reference: VersionMap
    private var cachedVersion = VersionMap()
                                ^
src_kt/java/arcs/storage/api/ArcsSet.kt:57:16: error: unresolved reference: Actor
    val actor: Actor = "ArcsSet@${hashCode()}"
               ^
src_kt/java/arcs/storage/api/ArcsSet.kt:105:41: error: cannot access class 'arcs.crdt.internal.VersionMap'. Check your module classpath for missing or conflicting dependencies
                cachedVersion = crdtSet.versionMap
```

**Comment by csilvestrini** [comment]
Rolling back in https://github.com/PolymerLabs/arcs/pull/4041, sorry Jason

**Comment by csilvestrini** [comment]
False alarm I think, never mind

**Pull request merged**